api/build.gradle

```diff
@@ -9,17 +9,42 @@ plugins {
 
 description = 'gRPC: API'
 
+sourceSets {
+    context {
+        java {
+            srcDir "${projectDir}/context/src/main/java"
+        }
+    }
+    main {
+        output.classesDirs.from(sourceSets.context.output.classesDirs)
+    }
+    test {
+        java {
+            srcDir "${projectDir}/context/src/test/java"
```

Comment:
Just put the tests in src/test/java (not under context)? It doesn't seem we're using the separate folder for anything, as the tests are compiled/run with all the other tests.